lib/config.js

```diff
@@ -44,6 +39,14 @@ Config.FILENAME = 'firebase.json';
 Config.TARGETS = ['hosting', 'rules'];
 Config.LEGACY_HOSTING_KEYS = ['public', 'rewrites', 'redirects', 'headers', 'ignore'];
 
+Config.prototype.importLegacyKeys = function() {
+  Config.LEGACY_HOSTING_KEYS.forEach(function(key) {
+    if (this._src[key]) {
```

Comment:
Since the legacy keys are well-defined and none accept a falsy value, I don't think it's strictly necessary, but can't hurt! Fixed.